### PR TITLE
Toolbar draging, closes #556

### LINKED ISSUE #556: When moving from compact to 'full' view you can move the draggable region off screen
To reproduce:
 - Select compact view
 - Move to the far left of the screen
 - Toggle back to full view

That window is now stuck!

Maybe we need to be more 'generous' with what is and is not draggable? Perhaps make the 'dead' areas around the title bar icons draggable?

### PR DESCRIPTION
**Pull request by bjedrzejewski**
Closes #556

### CODE REVIEW

**Comment by WPFerg** [comment]
Although it seems to be fine on normal view, it causes some unintended behaviour in compact (ie. top half of the icons not working, and making it very difficult to tearout the first stock:
image
image
image

**Comment by bjedrzejewski** [comment]
@WPFerg thanks for noticing, I forgot to test in compact, will fix it.

File: src/toolbar/toolbar.less [hunk 1]
```diff
@@ -1,14 +1,46 @@
...
+  .drag-bottom {
+    position: fixed;
+    height: @toolbar-icons-margin;
+    right: 0px;
+    top: (@toolbar-height - @toolbar-icons-margin);
+    height: @toolbar-icons-margin;
+    width: @sidebar-width;
   }
 
+
```
Comment (akhmorgan) [inline src/toolbar/toolbar.less]: white-space pedantry triggered.

**Comment by akhmorgan** [comment]
Looks good apart from the same comment as @WPFerg about issues in compact view. Will re-test when a fix is in for that.

**Comment by bjedrzejewski** [comment]
@WPFerg and @akhmorgan - ready to retest when you have a moment :)

**Comment by akhmorgan** [comment]
Looks good to me.

**Comment by WPFerg** [comment]
Better, though `.drag-left` is over the Closed Window List icon
image

**Comment by bjedrzejewski** [comment]
@WPFerg again, thanks for testing, need to come up with a clean solution for that.

**Comment by bjedrzejewski** [comment]
@WPFerg it works with the icon now :)

**Comment by bjedrzejewski** [comment]
Also, this could be written cleaner if the whole toolbar was smartly positioned.

File: src/toolbar/toolbar.less [hunk 2]
```diff
@@ -1,12 +1,41 @@
...
+    right: 0px;
+    width: @sidebar-width;
+  }
+
+  .drag-bottom {
+    position: fixed;
+    height: @toolbar-icons-margin;
+    right: 0px;
+    top: (@toolbar-height - @toolbar-icons-margin);
+    width: @sidebar-width;
```
Comment (owennw) [inline src/toolbar/toolbar.less]: I smell a mixin... !

File: src/toolbar/toolbar.less [hunk 3]
```diff
@@ -1,12 +1,41 @@
...
+    position: fixed;
+    height: @toolbar-height;
+    right: @toolbar-icons-width;
+    left: @sidebar-width;
+  }
+
+  .drag-right {
+    position: fixed;
+    height: @toolbar-height;
+    right: 0;
```
Comment (owennw) [inline src/toolbar/toolbar.less]: There's a dropped 'px' here, but that should be sorted by the mixin :)

**Comment by bjedrzejewski** [comment]
@owennw did my best with mixins here. Ready for review/merge if all good.

**Comment by bjedrzejewski** [comment]
Yeah, it was wrong in a first place. It needs some smart refactoring to make it not hardcoded. This is a part of #638

**Pull request merged**